test/models/setting_test.rb

```diff
@@ -667,11 +667,21 @@ def sticky_setting
   end
 
   test 'orders settings alphabetically' do
-    FactoryBot.create(:setting, :name => 'b_foo', :default => 'whatever', :value => 'whatever',  :full_name => 'B Foo Name')
-    FactoryBot.create(:setting, :name => 'a_foo', :default => 'whatever', :value => 'whatever',  :full_name => 'A Foo Name')
-    FactoryBot.create(:setting, :name => 'c_foo', :default => 'whatever', :value => 'whatever',  :full_name => 'C Foo Name')
-    settings = Setting.live_descendants.map(&:full_name).compact
-    assert_equal settings.sort, settings
+    a_name = 'a_foo'
+    b_name = 'b_foo'
+    c_name = 'c_foo'
+    FactoryBot.create(:setting, :name => b_name, :default => 'whatever', :value => 'whatever',  :full_name => 'B Foo Name')
+    FactoryBot.create(:setting, :name => a_name, :default => 'whatever', :value => 'whatever',  :full_name => 'A Foo Name')
+    FactoryBot.create(:setting, :name => c_name, :default => 'whatever', :value => 'whatever',  :full_name => 'C Foo Name')
+    settings = Setting.live_descendants.select(&:full_name)
+    a_foo = settings.index { |item| item.name == a_name }
+    b_foo = settings.index { |item| item.name == b_name }
+    c_foo = settings.index { |item| item.name == c_name }
+    assert a_foo
+    assert b_foo
+    assert c_foo
+    assert a_foo < b_foo
+    assert b_foo < c_foo
   end
```

Comment:
can you explain why this was changed?

Reply:
Long story short:

```ruby
settings = Setting.live_descendants.map(&:full_name).compact
#=> ["Access unattended without build", "A Foo Name", "B Foo Name", "C Foo Name"]
settings.sort
#=> ["A Foo Name", "Access unattended without build", "B Foo Name", "C Foo Name"]
```

By adding proper `setting_type` to one of the [fixtures](https://github.com/theforeman/foreman/pull/7411/files#diff-44b5b50a5d1889681cb7411ca971905a) I exposed a behaviour that made the test fail. Apparently, using order on postgres gives [surprising results](https://stackoverflow.com/questions/34537982/incorrect-sort-collation-order-with-spaces-in-postgresql-9-4) when whitespace is involved.